src/factory.rs

```diff
@@ -181,20 +200,49 @@ impl Factory {
     }
 
     fn mesh_vertices(shape: &GeometryShape) -> Vec<Vertex> {
-        if shape.normals.is_empty() {
-            shape.vertices.iter().map(|v| Vertex {
-                pos: [v.x, v.y, v.z, 1.0],
-                uv: [0.0, 0.0], //TODO
-                normal: NORMAL_Z,
-            }).collect()
-        } else {
-            let f2i = |x: f32| I8Norm(cmp::min(cmp::max((x * 127.) as isize, -128), 127) as i8);
-            shape.vertices.iter().zip(shape.normals.iter()).map(|(v, n)| Vertex {
-                pos: [v.x, v.y, v.z, 1.0],
-                uv: [0.0, 0.0], //TODO
-                normal: [f2i(n.x), f2i(n.y), f2i(n.z), I8Norm(0)],
-            }).collect()
-        }
+        let position_iter = shape.vertices.iter();
+        let normal_iter: Box<Iterator<Item = [gfx::format::I8Norm; 4]>> = {
```

Comment:
Would [`itertools::Either`](https://docs.rs/itertools/0.6.1/itertools/enum.Either.html) be a good alternative? We would have to rely on the compiler moving the loop condition out of the loop. There are a lot of branches to consider otherwise.